logger/logger.go

```diff
@@ -93,3 +93,16 @@ func New(o *Options) *slog.Logger {
 
 	return logger
 }
+
+type MigrationLogger struct {
```

Comment:
Can we move this to goose.go (or even to a separate pkg)?

I propose to call it `GooseLogAdapter`, as that's what it's for, right?

Reply:
https://github.com/pressly/goose/pull/989#pullrequestreview-3238236127 🎉